core/common/src/grpc_service.rs

```diff
@@ -0,0 +1,11 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+// SPDX-License-Identifier: MIT
+
+use tonic::transport::server::RoutesBuilder;
+
+/// Trait that must be implemented for a service to add a grpc service to the hosted server.
+pub trait GrpcService {
+    /// Function to add necessary services to the server builder.
+    fn add_grpc_services(&self, builder: &mut RoutesBuilder);
```

Comment:
Renamed the trait to GrpcExtension to be more clear its a higher level trait than a single grpc server service. This may be renamed to something like `module`